pandas/core/groupby/generic.py

```diff
@@ -1605,19 +1608,19 @@ def _python_agg_general(self, func, *args, **kwargs):
         res.columns = obj.columns.copy(deep=False)
         return self._wrap_aggregated_output(res)
 
-    def _aggregate_frame(self, func, *args, **kwargs) -> DataFrame:
+    def _aggregate_frame(self, func, *args, numeric_only=False, **kwargs) -> DataFrame:
         if self._grouper.nkeys != 1:
             raise AssertionError("Number of keys must be 1")
 
-        obj = self._obj_with_exclusions
-
+        mgr = self._get_data_to_aggregate(numeric_only=numeric_only)
+        data = self._wrap_agged_manager(mgr)
```

Comment:
I don't see any advantage (or disadvantage - for that matter) of calling this obj vs data. Because of this, I think we should leave it as obj (to avoid the code churn).

If you see a reason to prefer data over obj, let me know!

Reply:
yeah, the issue I was having on this point was because `_obj_with_exclusions` doesn't update when the dataframe is updated to remove non-numeric columns. I assume this is related to it being a `@cache_readonly` in `pandas.core.base.py`. My tentative workaround was to generate the data again. 

On my local branch, I simplified this by creating another object:

```
@cache_readonly
def _obj_numeric_only_with_exclusions(self):
```

and in those segments:

```
if numeric_only:
    obj = self._obj_numeric_only_with_exclusions(self):
else:
    obj = self._obj_with_exclusions
```

Do you have any thoughts on this approach?